### PR TITLE
Adding install for package and renaming project

### PR DESCRIPTION
**Pull request by fabiolrdiniz**
## Description
This PR is adding the install procedures for this repo to behave like a package, allowing one to rely on the `ioda-data_FOUND` variable in a bundle. It's also renaming the project to be consistent with its repo name (with a dash instead of an underscore).

Note: unfortunately, the ecbuild-based option for it is being added here, due to my lack of knowledge of the minimal lines to be added without that dependency --- I believe its removal can be done later when removing all the ecbuild dependencies across the repos.

## Issue(s) addressed
TO-BE-CREATED

## Dependencies
None.

## Impact

Expected impact on downstream repositories:

## Checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have run the unit tests before creating the PR

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -9,5 +9,7 @@
 
 cmake_minimum_required( VERSION 3.3.2 FATAL_ERROR )
 
-project( ioda_data VERSION 2.7.0 DESCRIPTION "IODA Test Files" )
+project( ioda-data VERSION 2.7.0 DESCRIPTION "IODA Test Files" )
```
Comment (srherbener) [inline CMakeLists.txt:12]: I think we should not use the hyphen "-" in the project name. The reason is that you cannot use a hyphen in a shell environment variable name. Ie the recommendation about setting "ioda-data_DIR" variable is not possible to do:
```
CMake Error at ioda/CMakeLists.txt:76 (find_package):
  By not providing "Findioda-data.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "ioda-data", but CMake did not find one.

  Could not find a package configuration file provided by "ioda-data" with
  any of the following names:

    ioda-dataConfig.cmake
    ioda-data-config.cmake

  Add the installation prefix of "ioda-data" to CMAKE_PREFIX_PATH or set
  "ioda-data_DIR" to a directory containing one of the above files.  If
  "ioda-data" provides a separate development package or SDK, be sure it has
  been installed.
```

**Review by srherbener — CHANGES_REQUESTED** [review]

**Comment by fabiolrdiniz** [comment]
Please, do not merge this until we decide on how to proceed. Changing it to be a draft.

**Comment by srherbener** [comment]
@fabiolrdiniz should we temporarily close this one too (since you closed https://github.com/JCSDA-internal/ioda/pull/1140)? Thanks!

**Comment by fabiolrdiniz** [comment]
@srherbener, sure!

**Comment by srherbener** [comment]
Thanks @fabiolrdiniz!

**Pull request closed without merging**